File: src/LianBiaoCode1.java
public class LianBiaoCode1 {

    //给一个 k 值 要求一个链表在 k值为一组的 范围内 翻转 不满k值一组的 不操作
    public class Node {
        public int value;
        public Node next;
    }

    public static Node getlastFromK(Node n, int k) {
        while (k > 0) {
            if (n != null) {
                n = n.next;
            }
            k--;
        }
        return n;
    }

    public static void fanZhuan(Node head, Node last) {
        last = last.next;
        Node pre = null;
        Node cur = head;
        Node next = null;
        while (cur != last) {

            next = cur.next;
            cur.next = pre;
            pre = cur;
            cur = next;
        }

        head.next = last;


    }


    public static void main(String[] args) {
        int a = 5;
        while (a >= 0) {
            System.out.println(a);
            a--;
        }

    }
}

File: src/python/lian_biao_code1.py
class LianBiaoCode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseKGroup(self, head, k):
        start = head
        one = self.getLast(head, k)
        if one is None:
            return head
        head = one
        self.rever(start, one)
        last = start
        while last.next is not None:
            start = last.next
            end = self.getLast(start, k)
            if end is None:
                return head
            self.rever(start, end)
            last.next = end
            last = start
        return one

    def getLast(self, head, k):
        while k - 1 > 0:
            if head is not None:
                head = head.next

        return head

    def rever(self, head, end):
        end = end.next
        pre = None
        cur = head
        last = None
        while cur != end:
            last = cur.next
            cur.next = pre
            pre = cur
            cur = last
        head.next = end


